src/_guides/language/sound-problems.md

```diff
@@ -466,29 +487,46 @@ filterValues((x) => (x [!as String!]).contains('Hello'));
 
 <hr>
 
-<a id="common-errors-and-warnings"></a>
-## Runtime errors
+### Incorrect type inference
 
-Dart enforces a sound type system. Roughly, this means
-you can't get into a situation where the value stored in a variable is
-different from the variable's static type. Like most modern statically
-typed languages, Dart accomplishes this with a combination of static
-(compile-time) and dynamic (runtime) checking.
+```nocode
+error - The return type '...' isn't a '...', as required by the closure's context - return_of_invalid_type_from_closure
+```
 
-For example, the following type error is detected at compile-time:
+On rare occasions, Dart's enhanced type inference might infer
+the wrong type for function literal arguments in a generic constructor invocation.
+This primarily affects `Iterable.fold`.
+
+#### Example
+
+In the following code,
+type inference will infer that `a` has a type of `Null`:
 
 {:.fails-sa}
+<?code-excerpt "lib/common_fixes_analysis.dart (type-inf-null)"?>
```

Comment:
You'll need to rerun the code excerpt tool again after making changes to the excerpt files.